packages/cactus-test-tooling/src/main/typescript/substrate-test-ledger/substrate-test-ledger.ts

```diff
@@ -0,0 +1,202 @@
+      },
+      HostConfig: {
+        AutoRemove: true,
+        PublishAllPorts: this.publishAllPorts,
+        Privileged: true,
+        PortBindings: {
+          "9944/tcp": [{ HostPort: "9944" }],
+        },
+      },
+    };
+
+    this.log.debug(`Starting ${this.imageFqn} with options: `, createOptions);
+
+    return new Promise<Container>((resolve, reject) => {
+      const eventEmitter: EventEmitter = docker.run(
+        this.imageFqn,
+        [],
+        [],
+        createOptions,
+        {},
+        (err: Error) => {
+          if (err) {
+            const errorMessage = `Failed to start container ${this.imageFqn}`;
+            const exception = new RuntimeError(errorMessage, err);
+            this.log.error(exception);
+            reject(exception);
+          }
+        },
+      );
+
+      eventEmitter.once("start", async (container: Container) => {
+        const { id } = container;
+        this.log.debug(`Started ${this.imageFqn} successfully. ID=${id}`);
+        this._containerId = Optional.ofNonNull(id);
+
+        if (this.emitContainerLogs) {
+          const logOptions = { follow: true, stderr: true, stdout: true };
+          const logStream = await container.logs(logOptions);
+          logStream.on("data", (data: Buffer) => {
+            const fnTag = `[${this.imageFqn}]`;
+            this.log.debug(`${fnTag} %o`, data.toString("utf-8"));
+          });
+        }
+        this.log.debug(`Registered container log stream callbacks OK`);
+
+        try {
+          this.log.debug(`Starting to wait for healthcheck... `);
+          await Containers.waitForHealthCheck(this.containerId.get());
+          this.log.debug(`Healthcheck passed OK`);
+          resolve(container);
+        } catch (ex) {
+          this.log.error(ex);
+          reject(ex);
+        }
+      });
+    });
+  }
+
+  public async stop(): Promise<unknown> {
+    return Containers.stop(this.container.get());
+  }
+
+  public async destroy(): Promise<unknown> {
+    return this.container.get().remove();
+  }
+
+  public async getContainerIpAddress(): Promise<string> {
+    const containerInfo = await this.getContainerInfo();
+    return Containers.getContainerInternalIp(containerInfo);
+  }
+
+  protected async getContainerInfo(): Promise<ContainerInfo> {
+    const fnTag = "FabricTestLedgerV1#getContainerInfo()";
+    const docker = new Docker();
+    const image = this.getContainerImageName();
+    const containerInfos = await docker.listContainers({});
+
+    let aContainerInfo;
+    if (this.containerId !== undefined) {
+      aContainerInfo = containerInfos.find(
+        (ci) => ci.Id == this.containerId.toString(),
+      );
+    }
+
+    if (aContainerInfo) {
+      return aContainerInfo;
+    } else {
+      throw new Error(`${fnTag} no image "${image}"`);
+    }
+  }
+
+  // ./scripts/docker_run.sh ./target/release/node-template purge-chain --dev
+  protected async purgeDevChain(): Promise<void> {
+    throw new Error("TODO");
```

Comment:
@RafaelAPB @CatarinaPedreira Is this still TODO in the scope of this PR or meant to be implemented in a future PR?

Reply:
I believe this can be done at a later stage

Reply:
Yes, I believe so too